dicom_utils/types.py

```diff
@@ -12,108 +13,138 @@
     UNKNOWN = 0
     NORMAL = 1
     SVIEW = 2
     TOMO = 3
 
     @staticmethod
-    def from_dicom(dcm: Dicom) -> "ImageType":
-        keys = process_image_type(dcm)
-        return get_simple_image_type(keys)
+    def from_dicom(dcm: Dicom) -> "SimpleImageType":
+        img_type = ImageType.from_dicom(dcm)
+        return img_type.to_simple_image_type()
 
     def __str__(self) -> str:
-        if self is ImageType.UNKNOWN:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        if self is self.UNKNOWN:
             return "unknown"
-        elif self is ImageType.NORMAL:
+        elif self is self.NORMAL:
             return "2d"
-        elif self is ImageType.SVIEW:
+        elif self is self.SVIEW:
             return "s-view"
-        elif self is ImageType.TOMO:
+        elif self is self.TOMO:
             return "tomo"
         else:
             raise RuntimeError("unknown ImageType value")
 
 
-def get_simple_image_type(image_type: Dict[str, Any]) -> ImageType:
-    r"""Produces an ImageType value from a dictionary of processed image type
-    fields (from :func:`process_image_type`).
+@dataclass(frozen=True)
+class ImageType:
+    """Container for DICOM metadata fields related to the image type.
+
+    Contains the following attributes:
 
-    ImageType is determined as follows:
-        1. ImageType.TOMO if ``NumberOfFrames > 1``
-        2. ImageType.SVIEW if ``pixels == "derived`` and (``"generated" in extras or flavor is not None``)
-        3. ImageType.2D otherwise
+        * ``"pixels"`` - First element of the IMAGE_TYPE field.
+        * ``"exam"`` - Second element of the IMAGE_TYPE field.
+        * ``"flavor"`` - Third element of the IMAGE_TYPE field.
+        * ``"extras"`` - Additional IMAGE_TYPE fields if available
+        * ``"NumberOfFrames"`` - Frame count (for TOMO images only)
+        * ``"model"`` - Manufacturer's model name
     """
-    # if fields 1 and 2 were missing, we know nothing
-    if not image_type:
-        return ImageType.UNKNOWN
-
-    pixels = image_type.get("pixels", "").lower() or ""
-    flavor = image_type.get("flavor", "").lower() or ""
-    exam = image_type.get("exam", "").lower() or ""
-    extras = image_type.get("extras", []) or []
-    num_frames = int(image_type.get("NumberOfFrames", 1) or 1)
-    machine = image_type.get("model", "").lower() or ""
-
-    # very solid rules
-    if num_frames > 1:
-        return ImageType.TOMO
-    if "original" in pixels:
-        return ImageType.NORMAL
-
-    # ok rules
-    if extras is not None and any("generated_2d" in x.lower() for x in extras):
-        return ImageType.SVIEW
-
-    # not good rules
-    if pixels == "derived" and exam == "primary" and machine == "fdr-3000aws" and flavor != "post_contrast":
-        return ImageType.SVIEW
-
-    return ImageType.NORMAL
+
+    pixels: str
+    exam: str
+    flavor: Optional[str] = None
+    extras: Optional[List[str]] = None
+    NumberOfFrames: Optional[int] = None
```

Comment:
Since you pull NumberOfFrames with `dcm.get("NumberOfFrames", 1)`, does NumberOfFrames need to be optional, or could we always store it as 1 or >1?

Reply:
I think it would be better to pull with `dcm.get("NumberOfFrames", None)` so that we accurately represent what was present in the metadata.

Reply:
Makes sense. I'm fine with that too.